app/services/registration_csv_converter.rb

```diff
@@ -49,7 +49,17 @@ def initialize(csv_string:, params:)
   #  14: tags (optional, may repeat)
 
   def convert
-    CSV.parse(csv_string, headers: true).map { |row| convert_row(row) }
+    csv_table = CSV.parse(csv_string, headers: true)
+    remove_blank_rows_at_end(csv_table)
+    csv_table.map { |row| convert_row(row) }
+  end
+
+  def remove_blank_rows_at_end(csv_table)
+    last = csv_table.size - 1
+    while csv_table[last].fields.all?(&:blank?) && last != 0
+      csv_table.delete(last) if csv_table[last].fields.all?(&:blank?)
+      last -= 1
+    end
```

Comment:
AFAICT, we can get here without going through the upload validator so it is needed.  However, if a csv file is uploaded on the registration form, the validation needs to happen in THAT class or there will be an error on the last (blank) lines that is spurious.